ee/tabby-webserver/src/service/email/test_utils.rs

```diff
@@ -0,0 +1,43 @@
+use std::time::Duration;
+
+use tabby_db::DbConn;
+use tokio::process::{Child, Command};
+
+use crate::schema::email::{AuthMethod, EmailService, EmailSettingInput, Encryption};
+
+use super::new_email_service;
+
```

Comment:
Consider extract this into a struct like
```
struct TestEmailServer;

impl TestEmailServer {
  fn new();
  fn start();
  fn kill();
}

...
```

Reply:
Looking at it now, not sure this would benefit much. We always start it immediately after creating it, and it's killed on drop, so it would kind of boil down to a single function. I'll try to improve the structure of these functions.

Reply:
it could also bundle the methods for checking email / title / content / ...

Reply:
SGTM!